src/efi.rs

```diff
@@ -105,6 +110,70 @@ impl Efi {
     }
 }
 
+/// Convert a nul-terminated UTF-16 byte array to a String.
+fn string_from_utf16_bytes(slice: &[u8]) -> String {
+    // For some reason, systemd appends 3 nul bytes after the string.
+    // Drop the last byte if there's an odd number.
+    let size = slice.len() / 2;
+    let v: Vec<u16> = (0..size)
+        .map(|i| u16::from_ne_bytes([slice[2 * i], slice[2 * i + 1]]))
```

Comment:
I think we can use https://doc.rust-lang.org/std/primitive.slice.html#method.chunks here

Reply:
Oh, that does look nicer. My rust noobness shows through.

Reply:
Your code is great honestly!  Learning a language (programming or not) is always a journey; even the relatively small `std` in Rust has a lot of surface area and idioms.